docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+
+### Pre-Migration Testing
+
+1. **Test in non-production environment:**
+   ```bash
+   # Create test TemporalWorkerDeployment
+   kubectl apply -f test-worker.yaml
+   
+   # Verify worker registration
+   kubectl logs -l temporal.io/worker-deployment=test-worker
+   ```
+
+2. **Validate environment variables:**
+   ```bash
+   kubectl exec -it deployment/test-worker-v1 -- env | grep TEMPORAL
+   ```
+
+### Post-Migration Validation
+
+1. **Check version status:**
+   ```bash
+   kubectl get temporalworkerdeployment -o wide
+   ```
+
+2. **Monitor version transitions:**
+   ```bash
+   kubectl get events --field-selector involvedObject.kind=TemporalWorkerDeployment
+   ```
+
+3. **Validate workflow routing:**
+   - Start test workflows
+   - Verify they're routed to correct versions
+   - Check Temporal UI for version distribution
+
+## Common Migration Patterns
+
+Here are common scenarios when migrating from unversioned to versioned workflows:
+
+### Pattern 1: Microservices Architecture
+
+If you have multiple services with their own workers, migrate each service independently:
+
+```yaml
+# Payment service worker
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: payment-processor
+  namespace: payments
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: payments
+  cutover:
+    strategy: Progressive  # Conservative rollout for financial operations
+    steps:
+      - rampPercentage: 5
+        pauseDuration: 10m
+      - rampPercentage: 25
+        pauseDuration: 15m
+  # ... rest of config
+---
+# Notification service worker (separate service, different risk profile)
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: notification-sender
+  namespace: notifications
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: notifications
+  cutover:
+    strategy: AllAtOnce  # Lower risk, faster rollouts acceptable
+  # ... rest of config
+```
+
+### Pattern 2: Environment-Specific Strategies
+
+Use different rollout strategies based on environment risk:
+
+```yaml
+# Production - Conservative rollout
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: order-processor
+  namespace: production
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  cutover:
+    strategy: Progressive
+    steps:
+      - rampPercentage: 10
+        pauseDuration: 15m
+      - rampPercentage: 50
+        pauseDuration: 30m
+    gate:
+      workflowType: "HealthCheck"  # Validate new version before proceeding
+---
+# Staging - Fast rollout for testing
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: order-processor
+  namespace: staging
+spec:
+  workerOptions:
+    connection: staging-temporal
+    temporalNamespace: staging
+  cutover:
+    strategy: AllAtOnce  # Immediate cutover for faster iteration
```

Comment:
maybe just "faster rollout"? I feel like iteration speed is dependent on so many factors, least of which is how long it takes to deploy one service to one env